Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -38,6 +41,50 @@ variable {R : Type*}
 
 namespace LaurentSeries
 
+section HasseDeriv
+
+/-- The Hasse derivative of Laurent series, as a linear map. -/
+@[simps]
+def hasseDeriv (R : Type*) {V : Type*} [AddCommGroup V] [Semiring R] [Module R V] (k : ℕ) :
+    LaurentSeries V →ₗ[R] LaurentSeries V where
+  toFun f := HahnSeries.ofSuppBddBelow (fun (n : ℤ) => (Ring.choose (n + k) k) • f.coeff (n + k))
+    (forallLTEqZero_supp_BddBelow _ (f.order - k : ℤ)
+    (fun _ h_lt ↦ by rw [coeff_eq_zero_of_lt_order <| lt_sub_iff_add_lt.mp h_lt, smul_zero]))
+  map_add' f g := by
+    simp only [ofSuppBddBelow, add_coeff', Pi.add_apply, smul_add]
+    exact rfl
+  map_smul' r f := by
+    ext
+    simp only [ofSuppBddBelow, smul_coeff, RingHom.id_apply]
+    rw [smul_comm]
+
+variable [Semiring R] {V : Type*} [AddCommGroup V] [Module R V]
+
+theorem hasseDeriv_coeff (k : ℕ) (f : LaurentSeries V) (n : ℤ) :
+    (hasseDeriv R k f).coeff n = Ring.choose (n + k) k • f.coeff (n + k) :=
+  rfl
+
+theorem hasseDeriv_add (k : ℕ) (f g : LaurentSeries V) : hasseDeriv R k (f + g) =
+    hasseDeriv R k f + hasseDeriv R k g := by
+  simp
+
+theorem hasseDeriv_smul (k : ℕ) (r : R) (f : LaurentSeries V) : hasseDeriv R k (r • f) =
+    r • (hasseDeriv R k f) := by
+  simp
+
+theorem hasseDeriv_coeff_add (k : ℕ) (f g : LaurentSeries V) (n : ℤ) :
+    (hasseDeriv R k (f + g)).coeff n = (hasseDeriv R k f).coeff n +
+     (hasseDeriv R k g).coeff n := by
+  simp
```

Comment:
This looks like it is probably not pulling its weight either since it is just a combination of [map_add](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Hom/Defs.html#map_add) and [LaurentSeries.add_coeff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/HahnSeries/Addition.html#HahnSeries.add_coeff)
```suggestion
```